### ISSUE TITLE
Add WAN Status and WAN IP Address

### ISSUE THREAD

**Issue by auanasgheps** [lukdwo/OpenWRT-collectd-MQTT-HA#10]
Hey there, this is not a real issue but I'm opening it to discuss this topic.

I wanted to get WAN Status and WAN IP address from the OpenWRT Router to Home Assistant.
Whilst the IP Address can be achieved via a native integration in HA, the WAN Status needs some trickery.

I found out that is possible and someone has already done it [here](https://community.home-assistant.io/t/openwrt-wan-connection-monitor/188160). I played with it, it works, but it's very rought.

I started to improve it and eventually I rewrote it entirely.

I think it would fit this "collection" (but I would call it lite-integration): it's still MQTT based, but uses a hotplug script.

If you fancy, I can create a PR so we can get these two sensors added as well!

**Comment by auanasgheps** [comment]
Hey @lukdwo any feedback on this? I've been working on it in my branch and I've almost completed it.

**Comment by lukdwo** [comment]
Hi, I'm sorry I was out of the loop for a while. It is a lot of opportunities to upgrade this collectd-MQQ-HA solution. For instance I wrote basic script which use speedtest.net CLI to check network speed:
![image](https://github.com/user-attachments/assets/26ff3f3b-cec8-45ee-bff2-5440e5795eed)

# Use speedtest on exact SRV and save to file
speedtest -s 60535 --accept-gdpr -f json-pretty > speedtest.json


# Download
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/download/bandwidth -m "$(($(cat speedtest.json | jq ".download.bandwidth")/125000))"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/download/bytes -m "$(($(cat speedtest.json | jq ".download.bytes")/1000000))"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/download/elapsed -m "$(($(cat speedtest.json | jq ".download.elapsed")/1000))"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/download/latency -m "$(cat speedtest.json | jq ".download.latency.iqm")"

# PacketLoss
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/packetloss -m "$(cat speedtest.json | jq ".packetLoss")"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/resulturl -m "$(cat speedtest.json | jq ".result.url")"

# Upload
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/upload/bandwidth -m "$(($(cat speedtest.json | jq ".upload.bandwidth")/125000))"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/upload/bytes -m "$(($(cat speedtest.json | jq ".upload.bytes")/1000000))"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USERNAME -P PASSWORD -r -t speedtest/upload/elapsed -m "$(($(cat speedtest.json | jq ".download.elapsed")/1000))"
mosquitto_pub -h 192.168.100.100 -p 1883 -u USER -P PASSWORD -r -t speedtest/upload/latency -m "$(cat speedtest.json | jq ".download.latency.iqm")"

It is up to you what would you like to achieve and what you need.
If you need any support please let me know.

BTW: The easiest way to get public IP is to sent result of `curl ifconfig.me` to mqtt.

**Comment by auanasgheps** [comment]
The speedtest feature is very interesting!

> It is up to you what would you like to achieve and what you need.
If you need any support please let me know.

My goal is to get all the useful information we can.
I used to run a Asus router which has an excellent integration with HA, OpenWRT has basically nothing.

This collection is the best I've found on the web and I'm committed to make it better.

You can see the work I've done [in this branch](https://github.com/auanasgheps/OpenWRT-collectd-MQTT-HA/tree/wan-sensors). In addition to the WAN sensors I've also done some documentation improvements. 

I'm still not ready, I've just created sensors to get the current WAN speed in Mbits, which is very useful to monitor Internet usage.

I will do a PR when ready.

I also think we should change the name to make it a bit more user friendly.

**Comment by dim6003** [comment]
Hi, seems interesting. Any progress ?

**Comment by auanasgheps** [comment]
Kind of. I need to finish up work in my branch.

**Comment by auanasgheps** [comment]
Hi @dim6003 your comment has pushed me to finish the works on my branch. 
This is now done and I've just opened a very nice PR #11

We could do much more but for now it's looking and working great!

**Comment by lukdwo** [comment]
@auanasgheps  thanks for your great improvements. Maybe we can try create direct HA integration into HACS together. I have never created this before but I will search more details about this.

**Comment by auanasgheps** [comment]
Yeah I was thinking about a HA integration on HACS. I'm not a dev, so I don't know what I really can achieve. I'll try to experiment in the future!
